presage-store-sled/src/lib.rs

```diff
@@ -109,6 +109,7 @@ impl SchemaVersion {
 }
 
 impl SledStore {
+    #[allow(unused_variables)]
```

Comment:
I guess either we can switch to providing an `impl` or `StoreCipher` like you suggested, or at least return an error if this `passphrase.is_some()` and the `encryption` feature is not enabled?